src/quicktube.js

```diff
@@ -0,0 +1,284 @@
+            this.videoEl.setAttribute('data-video-playing', true);
+            this.addActiveState();
+            setTimeout(this.onPlayerPercent.bind(this), 1000, event.target);
+        }
+
+        if (event.data === YT.PlayerState.PAUSED) {
+            this.removeActiveState();
+        }
+
+        const videoData = event.target.getVideoData();
+        let label = videoData.title;
+        // Get title of the current page
+        const pageTitle = document.title;
+
+        // TODO figure out what this is all doing and test it
+        if (this.options.trackAnalytics) {
+            if (event.data === YT.PlayerState.PLAYING && YT.gaLastAction === 'p') {
+                label = `Video Played - ${videoData.title}`;
+                trackEvent({
+                    event: 'youtube',
+                    eventCategory: 'Youtube Videos',
+                    eventAction: pageTitle,
+                    eventLabel: label,
+                });
+                YT.gaLastAction = '';
+            }
+
+            if (event.data === YT.PlayerState.PAUSED) {
+                label = `Video Paused - ${videoData.title}`;
+                trackEvent({
+                    event: 'youtube',
+                    eventCategory: 'Youtube Videos',
+                    eventAction: pageTitle,
+                    eventLabel: label,
+                });
+                YT.gaLastAction = 'p';
+            }
+        }
+
+        if (event.data === YT.PlayerState.ENDED) {
+            this.stopVideo();
+        }
+    }
+
+    // report the % played if it matches 0%, 25%, 50%, 75% or completed
+    onPlayerPercent(originalEvent) {
+        const event = originalEvent;
+        if (this.options.trackAnalytics) {
+            if (event.getPlayerState() === YT.PlayerState.PLAYING) {
+                const currenDuration = event.getDuration();
+                const currentTime = event.getCurrentTime();
+                let time;
+
+                if (currenDuration - currentTime <= 1.5) {
+                    time = 1;
+                } else {
+                    time = (Math.floor(currentTime / (currenDuration * 4)) / 4).toFixed(2);
+                }
+
+                if (!event.lastP || time > event.lastP) {
+                    const videoData = event.getVideoData();
+                    let label = videoData.title;
+                    // Get title of the current page
+                    const pageTitle = document.title;
+                    event.lastP = time;
+                    label = `${time * 100}% Video played - ${videoData.title}`;
+                    trackEvent({
+                        event: 'youtube',
+                        eventCategory: 'Youtube Videos',
+                        eventAction: pageTitle,
+                        eventLabel: label,
+                    });
+                }
+
+                if (event.lastP !== 1) {
+                    setTimeout(this.onPlayerPercent.bind(this), 1000, event);
```

Comment:
Don't use `setTimeout` to pass params in combination with a bind, do:

`setTimeout(this.onPlayerPercent.bind(this, event), 1000);`

or:

```js
setTimeout(() => {
    this.onPlayerPercent(event);
}, 1000);
```